src/ui/castleinternaljoystickswindows.pas

```diff
@@ -215,7 +224,8 @@ procedure TWindowsJoysticksBackend.Poll(const List: TJoystickList;
     if BackendInfo.Caps.wCaps and JOYCAPS_POVCTS > 0 then
       state.dwFlags := state.dwFlags or JOY_RETURNPOVCTS;
 
-    if joyGetPosEx( i, @state ) = 0 then
+    JoyError := joyGetPosEx( i, @state );
+    if JoyError = JOYERR_NOERROR then
```

Comment:
You silently ignore the case when JoyError is neither `JOYERR_NOERROR` nor `JOYERR_UNPLUGGED` now.

Better to make it

```
case JoyError of
  JOYERR_NOERROR:
    begin
       ...
    end;
  JOYERR_UNPLUGGED:
    begin
       ...
    end;
  else WritelnWarning('Joystick %s unexpected error %d', [Joystick.Info.Name, JoyError]);
end;
```